homeassistant/components/zwave_js/__init__.py

```diff
@@ -637,10 +643,12 @@ async def async_register_node_in_dev_reg(self, node: ZwaveNode) -> dr.DeviceEntr
                 dsk = provisioning_entry.dsk
                 dsk_identifier = (DOMAIN, f"provision_{dsk}")
 
-                # If the pre-provisioned device has the DSK identifier, update it
+                # If the pre-provisioned device has the DSK identifier, remove it
                 if dsk_identifier in preprovisioned_device.identifiers:
+                    new_identifiers = preprovisioned_device.identifiers.copy()
+                    new_identifiers.remove(dsk_identifier)
                     self.dev_reg.async_update_device(
-                        preprovisioned_device.id, new_identifiers=ids
+                        preprovisioned_device.id, new_identifiers=new_identifiers
```

Comment:
If there's a preprovisioned device entry I think we should update that entry with the node id identifier and not create a new device entry. The user may have customized the device entry, and we want to keep that customization when the device gets the node id.